moldesign/models/lammps_model.py

```diff
@@ -0,0 +1,280 @@
+        results = self._evaluate_molecule()
+        return results
+
+    #################################################
+    # "Private" methods for managing LAMMPS are below
+
+    def _evaluate_molecule(self):
+        # Recreate system if molecule changed
+        self.prep()
+
+        # Run for 0 fs to evaluate system
+        my_lmps = self.lammps_system
+        my_lmps.command("thermo_style one")
+        my_lmps.run(0)
+
+        # Evaluate potential energy and forces
+        pe = my_lmps.eval("pe")
+        force_array = []
+        for i in range(0, my_lmps.atoms.natoms):
+            force_array.append(my_lmps.atoms[i].force)
+        return {'potential_energy': pe * u.kcalpermol,
+                'forces': force_array * u.kcalpermol / u.angstrom}
+
+    # NOTE: Used by workflows to create lammps formatted data file
+    def get_lammps_data_path(self):
+        self.prep()
+
+        # initialize lammps object
+        return self._lammps_data_path
+    
+    def _create_lammps(self):
+
+        """
+        Create a LAMMPS system. Use MDT molecule object to construct a LAMMPS system
+
+        """
+        crd_filepath = '/tmp/tmp.crd'
+        top_filepath = '/tmp/tmp.top'
+
+        # Put amber parameters
+        self.mol.ff.amber_params.prmtop.put(top_filepath)
+        self.mol.ff.amber_params.inpcrd.put(crd_filepath)
+
+        # Add file paths to list of files
+        self.files.append(crd_filepath)
+        self.files.append(top_filepath)
+
+        # Ensure force fields are assigned 
+        if 'amber_params' not in self.mol.ff:
+            raise NotImplementedError('Assign forcefield parameters to the system')
+
+        # Store the PRMTOP file if ff are assigned and use it to create lammps molecule data
+
+        # initialize lammps object
+        lmp = lammps()
+        pylmp = PyLammps(ptr=lmp)
+
+        # Create temperary data file
+        predictable_filename = 'data.lammps_mol'
+        data_path = os.path.join(self.tmpdir, predictable_filename)
+
+        # Add file paths to list of files
+        self.files.append(data_path)
+        self._lammps_data_path = data_path
+
+        # Convert amber params files to lammps data
+        Convert_Amber_files(crd_filepath, top_filepath, data_path)
+
+        # Set basic parameters. Convert_Amber_files uses harmonic style
+        pylmp.command("units real")
+        pylmp.command("dimension 3")
+        pylmp.command("atom_style full")
+        pylmp.command("pair_style lj/charmm/coul/long 8.0 10.0 10.0")
+        pylmp.command("bond_style harmonic")
+        pylmp.command("angle_style harmonic")
+        pylmp.command("dihedral_style harmonic")
+        pylmp.command("improper_style harmonic")
+        pylmp.command("kspace_style pppm 0.0001")
+        pylmp.command("read_data " + data_path)
+
+        # Assign lammps system
+        self.lammps_system = pylmp
+
+@exports
+class LAMMPSInteractive(EnergyModelBase):
+
```

Comment:
Should the "pair_style lj/charmm/coul/long 8.0 10.0 10.0" be changed to "pair_style lj/charmm/coul/charmm/implicit 8.0 10.0" of is this for something else?